Source/WebKit/WebProcess/gtk/GtkSettingsManagerProxy.cpp

```diff
@@ -62,17 +63,25 @@ void GtkSettingsManagerProxy::applySettings(GtkSettingsState&& state)
     if (state.fontName)
         g_object_set(m_settings, "gtk-font-name", state.fontName->utf8().data(), nullptr);
 
-    if (state.xftAntialias)
+    if (state.xftAntialias) {
         g_object_set(m_settings, "gtk-xft-antialias", *state.xftAntialias, nullptr);
+        applyFontOptions();
+    }
 
-    if (state.xftHinting)
+    if (state.xftHinting) {
         g_object_set(m_settings, "gtk-xft-hinting", *state.xftHinting, nullptr);
+        applyFontOptions();
+    }
 
-    if (state.xftHintStyle)
+    if (state.xftHintStyle) {
         g_object_set(m_settings, "gtk-xft-hintstyle", state.xftHintStyle->utf8().data(), nullptr);
+        applyFontOptions();
+    }
 
-    if (state.xftRGBA)
+    if (state.xftRGBA) {
         g_object_set(m_settings, "gtk-xft-rgba", state.xftRGBA->utf8().data(), nullptr);
+        applyFontOptions();
```

Comment:
Instead fo applying font options on every change, we could add a variable `fontOptionsDidChange` set when any setting changes and then call `applyFontOptions()` once since it applies all the options.